src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs

```diff
@@ -321,6 +336,12 @@ internal class CSharpRecommendationService : AbstractRecommendationService
                     position: name.SpanStart,
                     container: symbol);
 
+                if (context.IsNamespaceDeclarationNameContext)
+                {
+                    return symbols.Where(s => s.IsNamespace() &&
```

Comment:
Duplicaed code with above.  Can you extract out a helper to do this?  Like "IsNonIntersectingNamespace"?